brew_cask_and_adopt_manual_installed_applications/brew_cask_and_adopt_manual_installed_applications.py

```diff
@@ -127,85 +195,107 @@ def is_default_apple_app(app_name):
 
-    for app in apps:
-        app_name = app.replace(".app", "").lower()
+    for app in installed_apps:
+        # Split path and extract app name only
+        app_name = os.path.splitext(app)[0].lower()
 
+        print("")
+
+        # Skip if application is an Apple app
         if is_default_apple_app(app_name):
-            print(f"Skipping default Apple app: {app_name}")
-            print()
+            print(f'Skipping default Apple app "{app_name}"')
             continue
 
-        print(f"Checking for {app_name}...")
+        print(f'Checking for "{app_name}"...')
 
-        # Check if managed by Homebrew (exact name and variations)
-        if is_managed_by_brew(app_name) or is_managed_by_brew(
-            app_name.replace(" ", "-")
-        ):
-            print(f"{app_name} is already installed and managed via Homebrew.")
-            print()
+        # Check for valid cask name or suggest alternative name
+        brew_cask_app_name = check_cask_available(app_name)
+
+        # No valid alternative chosen or skipped
+        if brew_cask_app_name is None:
+            print_colored(f'"{app_name}" is not available as a cask.', Colors.RED)
+            non_brew_managed_apps.append(app_name)
             continue
 
-        # Normalize app name for Homebrew
-        normalized_name = normalize_app_name(app_name)
-        if normalized_name:
+        # Application already managed by Homebrew
+        if is_managed_by_brew(brew_cask_app_name):
             print(
-                f"{colors.GREEN}Installing {normalized_name} with Homebrew...{colors.ENDC}"
+                f'"{brew_cask_app_name}" is already installed and managed via Homebrew.'
             )
-            install_or_adopt_app(normalized_name, install_dir)
-        else:
-            print(f"{app_name} is not available as a cask.")
-            not_found_apps.append(app_name)
+            continue
+
+        # For manually mode prompt to ask for adoption
+        if manually and not prompt_yes_no(
+            f'Do you want to adopt "{brew_cask_app_name}"?'
+        ):
+            continue
 
-        print()
+        # Try to install application with Homebrew
+        print(f'Trying to install "{brew_cask_app_name}" with Homebrew...')
+        if install_adopt_app(brew_cask_app_name, install_dir):
+            print_colored(
+                f'Installation of "{brew_cask_app_name}" succeeded!', Colors.GREEN
+            )
 
-    if not_found_apps:
-        print(f"\n{colors.RED}Applications not found as Homebrew casks:{colors.ENDC}\n")
-        for app in not_found_apps:
+    if non_brew_managed_apps:
+        print_colored("\nApplications not found as Homebrew casks:", Colors.RED)
+        for app in non_brew_managed_apps:
             print(app)
-        print(
-            "\nFor more available Homebrew Casks, visit https://formulae.brew.sh/cask/"
-        )
+        print("For more available Homebrew Casks, visit https://formulae.brew.sh/cask/")
```

Comment:
Could you add an empty line before this print statement? Currently, it looks a bit too cluttered IMO.